askrene getroutes produces a malformed response

### Issue and Steps to Reproduce
In regtest the command `lightning-cli getroutes` produces this output:
```
lightning-cli: Malformed response '63x17618/1 -> 19801029msat/220 15x485x17484/0 -> 19800000msat/120 122x485x35423/1 -> 19797031msat/40 122x323x35357/0 -> 19793073x3301x1646/0 -> 35694180msat/74 0x122x2044/1 -> 35679909msat/40 122x323x35357/0 -> 35672775msat/6"}}5x63x17618/1 -> 19801029msat/220 15x485x17484/0 -> 19800000msat/120 122x485x35423/1 -> 19797031msat/40 122x323x35357/0 -> 19793073msat/6"}}

{"jsonrpc":"2.0","method":"message","params":{"id":"cli:getroutes#170504","level":"info","message":"Flow 2/4: 35712991msat/218 0x3301x1646/0 -> 35694180msat/74 0x122x2044/1 -> 35679909msat/40 122x323x35357/0 -> 35672775msat/6"}}x90x29364/1 -> 19801029msat/364 15x63x17618/1 -> 19801029msat/220 15x485x17484/0 -> 19800000msat/120 122x485x35423/1 -> 19797031msat/40 122x323x35357/0 -> 19793073msat/6"}}

{"jsonrpc":"2.0","method":"message","params":{"id":"cli:getroutes#170504","level":"info","message":"Flow 2/4: 35712991msat/218 0x3301x1646/0 -> 35694180msat/74 0x122x2044/1 -> 35679909msat/40 122x323x35357/0 -> 35672775msat/6"}}'
```
To reproduce it is enough to run lightning on regtest using the compressed gossmap `./tests/data/gossip-store-2024-09-22.compressed`
and call getroutes from node `022d223620a359a47ff7f7ac447c85c46c923da53389221a0054c11c1e3ca31d59`
to node `032592ebeab65d2c20a60b6eec9bc1b80df9148d2f71d91a670fe7f4f3c25a5bb9`, ie.

```
lightning-cli -k getroutes \
        source=022d223620a359a47ff7f7ac447c85c46c923da53389221a0054c11c1e3ca31d59 \
        destination=032592ebeab65d2c20a60b6eec9bc1b80df9148d2f71d91a670fe7f4f3c25a5bb9 \
        amount_msat=100000sat \
        final_cltv=6 \
        layers="[]" \
        maxfee_msat=500sat
```

My guess is that somehow `plugin_notify_message` is mixing with `getroutes` response, are they racing for the same pipe output?
Commenting the `plugin_notify_messsage` line in `rq_log` function in `askrene.c` file removes the problem.

Yes. Sorry, in my tests I was mapping node 3301 to `022d223620a359a47ff7f7ac447c85c46c923da53389221a0054c11c1e3ca31d59`.
I am quite not able to reproduce it on pytest.
I see it happen using a bash script:
```
#!/bin/sh

# environment taken from contrib/startup_regtest.sh

export GOSSMAP_COMPRESS=./devtools/gossmap-compress

# Do the Right Thing if we're currently in top of srcdir.
if [ -z "$LIGHTNING_BIN" ] && [ -x cli/lightning-cli ] && [ -x lightningd/lightningd ]; then
	LIGHTNING_BIN=$(pwd)
fi

if [ -z "$LIGHTNING_BIN" ]; then
	# Already installed maybe?  Prints
	if ! type lightning-cli >/dev/null 2>&1 ; then
		echo lightning-cli: not found
		return 1
	fi
	if ! type lightningd >/dev/null 2>&1 ; then
		echo lightningd: not found
		return 1
	fi
	LCLI=lightning-cli
	LIGHTNINGD=lightningd
else
	LCLI="$LIGHTNING_BIN"/cli/lightning-cli
	LIGHTNINGD="$LIGHTNING_BIN"/lightningd/lightningd
	# This mirrors "type" output above.
fi

if [ -z "$LIGHTNING_DIR" ]; then
    # Default is to use the /tmp directory
    LIGHTNING_DIR=/tmp/askrene_benchmark/lightning
fi

if [ -z "$BITCOIN_DIR" ]; then
    BITCOIN_DIR=/tmp/askrene_benchmark/bitcoin
fi

# shellcheck disable=SC2153
if [ -z "$BITCOIN_BIN" ]; then
	# Already installed maybe?  Prints
	if ! type bitcoin-cli >/dev/null 2>&1 ; then
		echo bitcoin-cli: not found
		return 1
	fi
	if ! type bitcoind >/dev/null 2>&1 ; then
		echo bitcoind: not found
		return 1
	fi
	BCLI=bitcoin-cli
	BITCOIND=bitcoind
else
    BCLI="$BITCOIN_BIN"/bitcoin-cli
    BITCOIND="$BITCOIN_BIN"/bitcoind
fi

if [ -z "$COMPRESSED_GOSSMAP" ]; then
	COMPRESSED_GOSSMAP="./tests/data/gossip-store-2024-09-22.compressed"
fi


echo lightning-cli is "$LCLI"
echo lightningd is "$LIGHTNINGD"
echo lightning-dir is "$LIGHTNING_DIR"
export LCLI="$LCLI"
export LIGHTNINGD="$LIGHTNINGD"
export LIGHTNING_DIR="$LIGHTNING_DIR"

echo bitcoin-cli is "$BCLI"
echo bitcoind is "$BITCOIND"
echo bitcoin-dir is "$BITCOIN_DIR"
export BCLI="$BCLI"
export BITCOIND="$BITCOIND"
export BITCOIN_DIR="$BITCOIN_DIR"

echo compressed gossmap is "$COMPRESSED_GOSSMAP"
export COMPRESSED_GOSSMAP="$COMPRESSED_GOSSMAP"

if test ! -f "$COMPRESSED_GOSSMAP"; then
	echo Compressed gossmap file not found, aborting
	return 1
fi

# generate gossmap file
# load a single lightning node using predefined gossmap

wait_for_lightningd() {
	echo awaiting lightningd...
	for i in $(seq "5"); do
		if $LCLI --lightning-dir="$LIGHTNING_DIR" getinfo > /dev/null 2>&1; then
			break
		else
			sleep 1
		fi
	done
}

start_lightningd() {
	socket=7070
	mkdir -p "$LIGHTNING_DIR"
	# Node config
	cat <<- EOF > "$LIGHTNING_DIR/config"
	network=regtest
	log-level=debug
	log-file=$LIGHTNING_DIR/log
	addr=localhost:$socket
	allow-deprecated-apis=false
	developer
	dev-fast-gossip
	dev-bitcoind-poll=5
	experimental-dual-fund
	experimental-splicing
	funder-policy=match
	funder-policy-mod=100
	funder-min-their-funding=10000
	funder-per-channel-max=100000
	funder-fuzz-percent=0
	funder-lease-requests-only=false
	lease-fee-base-sat=2sat
	lease-fee-basis=50
	invoices-onchain-fallback
	EOF

	# Start the lightning nodes
	test -f "$LIGHTNING_DIR/lightningd-regtest.pid" || \
		"$LIGHTNINGD" "--network=regtest" "--lightning-dir=$LIGHTNING_DIR" "--bitcoin-datadir=$BITCOIN_DIR" "--database-upgrade=true" &
}

ln_cli(){
	$LCLI --lightning-dir="$LIGHTNING_DIR" $@
}

start_btc(){
	# Start bitcoind in the background
	mkdir -p "$BITCOIN_DIR"
	test -f "$BITCOIN_DIR/regtest/bitcoind.pid" || \
		"$BITCOIND" -datadir="$BITCOIN_DIR" -regtest -txindex -fallbackfee=0.00000253 -daemon

	# Wait for it to start.
	while ! "$BCLI" -datadir="$BITCOIN_DIR" -regtest ping 2> /tmp/null; do echo "awaiting bitcoind..." && sleep 1; done

	# Check if default wallet exists
	if ! "$BCLI" -datadir="$BITCOIN_DIR" -regtest listwalletdir | jq -r '.wallets[] | .name' | grep -wqe 'default' ; then
		# wallet dir does not exist, create one
		echo "Making \"default\" bitcoind wallet."
		"$BCLI" -datadir="$BITCOIN_DIR" -regtest createwallet default >/dev/null 2>&1
	fi

	# Check if default wallet is loaded
	if ! "$BCLI" -datadir="$BITCOIN_DIR" -regtest listwallets | jq -r '.[]' | grep -wqe 'default' ; then
		echo "Loading \"default\" bitcoind wallet."
		"$BCLI" -datadir="$BITCOIN_DIR" -regtest loadwallet default >/dev/null 2>&1
	fi

	# Kick it out of initialblockdownload if necessary
	if "$BCLI" -datadir="$BITCOIN_DIR" -regtest getblockchaininfo | grep -q 'initialblockdownload.*true'; then
		"$BCLI" -datadir="$BITCOIN_DIR" -regtest generatetoaddress 1 "$($BCLI -datadir="$BITCOIN_DIR" -regtest getnewaddress)" > /dev/null
	fi

	echo "Commands: "
	echo "	bt_cli"
}
bt_cli(){
	$BCLI -datadir="$BITCOIN_DIR" -regtest $@
}

generate_gossmap(){
	mkdir -p "$LIGHTNING_DIR/regtest"
	"$GOSSMAP_COMPRESS" decompress  \
		"$COMPRESSED_GOSSMAP" "$LIGHTNING_DIR/regtest/gossip_store"
}

NODEIDS=""

start_cln(){
	GENIDS=`generate_gossmap`
	readarray -t NODEIDS <<< "$GENIDS"
	start_lightningd
	wait_for_lightningd

	echo "Commands: "
	echo "	ln_cli"
}

stop_cln() {
	ln_cli stop
	sleep 1
	test ! -f "$LIGHTNING_DIR/lightningd-regtest.pid" || \
		(kill "$(cat "$LIGHTNING_DIR/lightningd-regtest.pid")"; \
		rm "$LIGHTNING_DIR/lightningd-regtest.pid")
}

stop_btc(){
	bt_cli stop
	sleep 1
	test ! -f "$BITCOIN_DIR/regtest/bitcoind.pid" || \
		(kill "$(cat "$BITCOIN_DIR/regtest/bitcoind.pid")"; \
		rm "$BITCOIN_DIR/regtest/bitcoind.pid")
}

start_ln(){
	start_btc
	start_cln
}

stop_ln(){
	stop_cln
	stop_btc
}

destroy_ln(){
	rm -rf $LIGHTNING_DIR
	rm -rf $BITCOIN_DIR
}

cleanup(){
	stop_ln
	destroy_ln
}

trap cleanup EXIT

echo Useful commands:
echo "  stop_ln: shutdown"
echo "  destroy_ln: remove ln directories"

start_ln

source=${NODEIDS[3301]}

for idx in "${!NODEIDS[@]}"; do
	destination=${NODEIDS[$idx]}
	echo ID: $idx, source=$source, destination=$destination
	result=`ln_cli -k getroutes source=$source destination=$destination amount_msat=100000sat final_cltv=6 layers="[]" maxfee_msat=500sat`
	if [ $idx -ge 500 ]; then
		break
	fi
done

stop_ln

# time_list=`cat "$LIGHTNING_DIR/log" | grep "notify msg info: minflow elapsed time" | awk '{print $(NF-1)}'`
# echo $time_list | python3 plot-perf.py

destroy_ln
```

--- a/cli/lightning-cli.c
+++ b/cli/lightning-cli.c
@@ -860,10 +860,8 @@ int main(int argc, char *argv[])
 		default:
 			if (handle_notify(resp, toks, notification_level,
 					  &last_was_progress)) {
-				/* +2 for \n\n */
-				size_t len = toks[0].end - toks[0].start + 2;
-				memmove(resp, resp + len, off - len);
-				off -= len;
+				memmove(resp, resp + toks[0].end, off - toks[0].end);
+				off -= toks[0].end;
 				jsmn_init(&parser);
 				toks[0].type = JSMN_UNDEFINED;
 				/* Don't force another read! */

